tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_multiple_snapshots.py

```diff
@@ -0,0 +1,414 @@
+        log.info("Add subvolumes for mirroring to remote location")
+        fs_mirroring_utils.add_path_for_mirroring(
+            source_clients[0], source_fs, subvol1_path
+        )
+        fs_mirroring_utils.add_path_for_mirroring(
+            source_clients[0], source_fs, subvol2_path
+        )
+        md5sum = "md5sum_script.sh"
+        target_clients[0].upload_file(
+            sudo=True,
+            src="tests/cephfs/cephfs_mirror_upgrade/md5sum_script.sh",
+            dst=f"/root/{md5sum}",
+        )
+        source_clients[0].upload_file(
+            sudo=True,
+            src="tests/cephfs/cephfs_mirror_upgrade/md5sum_script.sh",
+            dst=f"/root/{md5sum}",
+        )
+        snap1 = "snap_k1_mul"
+        snap2 = "snap_f1_mul"
+        source_clients[0].exec_command(
+            sudo=True,
+            cmd=f"mkdir -p {kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir};"
+            f"python3 /home/cephuser/smallfile/smallfile_cli.py "
+            f"--operation create --threads 1 --file-size 1 "
+            f"--files 5 --files-per-dir 1 --dirs-per-dir 2 --top "
+            f"{kernel_mounting_dir_1}/{subvol1_path}/{mounting_dir}",
+        )
+
+        source_clients[0].exec_command(
+            sudo=True,
+            cmd=f"mkdir {kernel_mounting_dir_1}{subvol1_path}.snap/{snap1}",
+        )
+        fsid = fs_mirroring_utils.get_fsid(cephfs_mirror_node[0])
+        log.info(f"fsid on ceph cluster : {fsid}")
+        daemon_names = fs_mirroring_utils.get_daemon_name(source_clients[0])
+
+        asok_files = fs_mirroring_utils.get_asok_file(
+            cephfs_mirror_node, fsid, daemon_names
+        )
+        log.info(f"Admin Socket file of cephfs-mirror daemon : {asok_files}")
+        filesystem_id = fs_mirroring_utils.get_filesystem_id_by_name(
+            source_clients[0], source_fs
+        )
+        log.info(f"filesystem id of {source_fs} is : {filesystem_id}")
+        peer_uuid = fs_mirroring_utils.get_peer_uuid_by_name(
+            source_clients[0], source_fs
+        )
+        log.info(f"peer uuid of {source_fs} is : {peer_uuid}")
+
+        for node in cephfs_mirror_node:
+            node.exec_command(sudo=True, cmd="yum install -y ceph-common --nogpgcheck")
+
+        fs_mirroring_utils.validate_snapshot_sync_status(
+            cephfs_mirror_node,
+            source_fs,
+            snap1,
+            fsid,
+            asok_files,
+            filesystem_id,
+            peer_uuid,
+        )
+        out, rc = fs_mirroring_utils.list_and_verify_remote_snapshots_and_data_checksum(
+            target_clients[0],
+            "client.admin",
+            subvol1_path,
+            source_clients[0],
+            kernel_mounting_dir_1,
+            target_fs,
+        )
+        if not out:
+            raise CommandFailed(f"{rc}")
+        log.info(
+            f"{snap1} was synced even after disconnecting the mirror daemon node for 90 seconds "
+        )
+        log.info(
```

Comment:
can we move the logger message to line 251 just before removing the peer connection ?